src/org/project/Order.java

```diff
@@ -13,12 +13,15 @@ public class Order {
     private List<Product> products;
     private BigDecimal totalPrice;
     private LocalDateTime orderTime;
+    private BigDecimal discountAmount = BigDecimal.ZERO;
+    private BigDecimal finalPrice;
```

Comment:
imo to troche miesza ze mamy totalPrice i finalPrice, albo trzeba zmienic nazwe ktorejs zmiennej na bardziej opisowy taki zeby bylo jasne ktora cena jest czym, albo dzialac na jednej cenie i zmieniac po prostu np total/final po zmienianie discount